modules/migration-configuring-aws-s3.adoc

```diff
@@ -0,0 +1,136 @@
+** The source and target clusters must be in the same region.
+** The source and target clusters must have the same storage class.
+** The storage class must be compatible with snapshots.
+
+.Procedure
+
+. Create an AWS S3 bucket:
++
+----
+$ aws s3api create-bucket \
+    --bucket <bucket_name> \ <1>
+    --region <bucket_region> <2>
+----
+<1> Specify your S3 bucket name.
+<2> Specify your S3 bucket region, for example, `us-east-1`.
+
+. Create the IAM user `velero`:
++
+----
+$ aws iam create-user --user-name velero
+----
+
+. Create an EC2 EBS snapshot policy:
++
+----
+$ cat > velero-ec2-snapshot-policy.json <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ec2:DescribeVolumes",
+                "ec2:DescribeSnapshots",
+                "ec2:CreateTags",
+                "ec2:CreateVolume",
+                "ec2:CreateSnapshot",
+                "ec2:DeleteSnapshot"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+----
+
+. Create an AWS S3 access policy for one or for all S3 buckets:
++
+----
+$ cat > velero-s3-policy.json <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetObject",
+                "s3:DeleteObject",
+                "s3:PutObject",
+                "s3:AbortMultipartUpload",
+                "s3:ListMultipartUploadParts"
+            ],
+            "Resource": [
+                "arn:aws:s3:::<bucket_name>/*" <1>
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListBucket",
+                "s3:GetBucketLocation",
+                "s3:ListBucketMultipartUploads"
+            ],
+            "Resource": [
+                "arn:aws:s3:::<bucket_name>" <1>
+            ]
+        }
+    ]
+}
+EOF
+----
+<1> To grant access to a single S3 bucket, specify the bucket name. To grant access to all AWS S3 buckets, specify `*` instead of a bucket name:
++
+----
+"Resource": [
+    "arn:aws:s3:::*"
+----
+
+. Attach the EC2 EBS policy to `velero`:
++
+----
+$ aws iam put-user-policy \
+  --user-name velero \
+  --policy-name velero-ebs \
+  --policy-document file://velero-ec2-snapshot-policy.json
+----
+
+. Attach the AWS S3 policy to `velero`:
++
+----
+$ aws iam put-user-policy \
+  --user-name velero \
+  --policy-name velero-s3 \
+  --policy-document file://velero-s3-policy.json
+----
+
+. Create an access key for `velero`:
++
+----
+$ aws iam create-access-key --user-name velero
+{
+  "AccessKey": {
+        "UserName": "velero",
+        "Status": "Active",
+        "CreateDate": "2017-07-31T22:24:41.576Z",
+        "SecretAccessKey": <AWS_SECRET_ACCESS_KEY>, <1>
+        "AccessKeyId": <AWS_ACCESS_KEY_ID> <1>
+    }
+}
+----
+<1> Record the `AWS_SECRET_ACCESS_KEY` and the `AWS_ACCESS_KEY_ID`.
```

Comment:
These few places where you say "Record", do you mean because they're going to be used in a later procedure? If so, I might make that clearer.

Reply:
That's a lot of places to mention it! OK, I'll think about it.